internal/repos/azuredevops.go

```
@@ -109,11 +116,21 @@ func (s *AzureDevOpsSource) processReposFromProjectOrOrg(ctx context.Context, na
 		return
 	}
 
-	for _, repo := range repos {
-		if s.exclude(fmt.Sprintf("%s/%s", repo.Project.Name, repo.Name)) {
+	for _, azureRepo := range repos {
+		if s.exclude(fmt.Sprintf("%s/%s", azureRepo.Project.Name, azureRepo.Name)) {
 			continue
 		}
-		repo, err := s.makeRepo(repo)
+
+		var repo *types.Repo
+		var err error
+		if callType == repoFromOrg {
+			repo, err = s.makeRepoFromOrg(azureRepo, name)
+		} else if callType == repoFromProject {
+			repo, err = s.makeRepoFromProject(azureRepo, name)
+		} else {
+			panic("processReposFromProjectOrg called with unsupported callType, must be org or project")
```

I'm not sure about how we treat `panic` usages, should we just log and return maybe?

---

Log and return would be a silent failure. Since it's an internal function and has a limited use-case, a panic ensures this fails during development if someone adds a third case in the call site but does not handle it here.

---

We use panic for such not-expected scenarios in other parts of the code.